### PR TITLE
Add enrollments to sessions

### PR DESCRIPTION
**Pull request by alxdarksage**
BRIDGE-3097

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java [hunk 1]
```diff
@@ -555,6 +558,41 @@ public void consentAndWithdrawFromAppUpdatesDataGroupsAndStudies() throws Except
...
+            Enrollment en = researchersApi.withdrawParticipant(STUDY_ID_1, user.getUserId(), "Withdrawn").execute().body();
+            
+            assertEquals(researchUser.getUserId(), en.getWithdrawnBy());
+            
+            ParticipantsApi participantsApi = researchUser.getClient(ParticipantsApi.class);
+
+            // verify the participant is now not enrolled in any study
+            StudyParticipant participant = participantsApi.getParticipantById(user.getUserId(), false).execute().body();
+            assertNull(participant.getEnrollments().get(STUDY_ID_1).isWithdrawnBySelf());
+            assertNotNull(researchUser.getUserId(), participant.getEnrollments().get(STUDY_ID_1).getWithdrawnOn());
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java:578]: These should be two separate tests

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java [hunk 2]
```diff
@@ -555,6 +558,41 @@ public void consentAndWithdrawFromAppUpdatesDataGroupsAndStudies() throws Except
...
+            researchersApi.withdrawParticipantFromApp(user.getUserId(), 
+                    new Withdrawal().reason("Withdrawn")).execute().body();
+            
+            ParticipantsApi participantsApi = researchUser.getClient(ParticipantsApi.class);
+
+            // verify the participant is now not enrolled in any study
+            StudyParticipant participant = participantsApi.getParticipantById(user.getUserId(), false).execute().body();
+            assertNull(participant.getEmail());
+            assertNull(participant.getEnrollments().get(STUDY_ID_1).isWithdrawnBySelf());
+            assertNotNull(researchUser.getUserId(), participant.getEnrollments().get(STUDY_ID_1).getWithdrawnOn());
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/ConsentTest.java:595]: These should be two separate tests

**Review by jnelsonjava — APPROVED** [review]
Approved with just the one comment to check out.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ParticipantsTest.java [hunk 3]
```diff
@@ -201,6 +206,11 @@ public void retrieveParticipant() throws Exception {
             assertEquals(user.getSession().getId(), participant.getId());
             assertFalse(participant.getConsentHistories().isEmpty());
             
+            assertEquals(user.getSession().getEnrollments().get(STUDY_ID_1).getEnrolledOn(), 
+                    participant.getEnrollments().get(STUDY_ID_1).getEnrolledOn());
+            assertEquals(user.getSession().getEnrollments().size(), participant.getEnrollments().size());
+            assertEquals((String)null, (String)null);
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ParticipantsTest.java:212]: Sorry that was a test to see if asserEquals would return true if both values were null, which apparently it does. So it alerted me to the fact that a test like assertEquals(foo.getValue(), bar.getValue()) might succeed even though the value isn't there. Removed.

**Pull request merged**